bazel/third-party/BUILD.curl.bazel

```diff
@@ -0,0 +1,115 @@
+# This file is copied from https://github.com/3rdparty/bazel-rules-curl/tree/f64665f4614beb19f08b45b755d7c77f487287bb
```

Comment:
I don't really understand, what is done here.  But is it using boringssl or openssl?  I guess it would make sense to keep these things consistent over the different build systems, i.e. either openssl or boringssl

Reply:
Bazel is a hermetic build system, i.e. it is building the world from scratch. These are just rules to build libcurl. This is using boringssl, and we only use boringssl everywhere.

Reply:
My understanding is that the "libcurl" normally comes from the system, and is not predefined. So it's up to system/distribution maintainer to choose how libcurl is build.

Here in the bazel build we have to make this choice as well. I've chose boringssl, because it seems to be more bazel-friendly (namely they have bazel support built-in).

Reply:
The main features in core for e.g. TPM based TLS for OCPP etc will use openssl3, no idea if this is supported on boringssl as well. Also google changes the API as they like, so I would not claim we officially support boringssl at all.

Reply:
I've added warning about BoringSSL to Readme.

We need more details regarding the requirements. I'm up to support openssl1 or 3 eventually.